src/js/plugins/plugin.text_selection.js

```diff
@@ -0,0 +1,179 @@
+//@ts-check
+const BookReader = /** @type {typeof import('../BookReader').default} */(window.BookReader);
+
+class TextSelectionPlugin {
+
+  constructor() {
+    /**@type {PromiseLike<JQuery<HTMLElement>>} */
+    this.djvuPagesPromise = null;
+  }
+
+  /**
+   * @param {string} ocaid
+   */
+  init(ocaid) {
+    this.djvuPagesPromise = $.ajax({
+      type: "GET",
+      url: `https://cors.archive.org/cors/${ocaid}/${ocaid}_djvu.xml`,
+      dataType: "xml",
+
+      error: function (e) {
+        return undefined;
+      }
+    }).then(function (response) {
+      const xmlMap = response;
+
+      if (xmlMap != undefined) {
+        return $(xmlMap).find("OBJECT");
+      }
+    });
+  }
+
+  /**
+   * @param {number} index
+   * @returns {Promise<HTMLElement>}
+   */
+  async getPageText(index) {
+    return (await this.djvuPagesPromise)[index];
+  }
+
+  /**
+   * @param {JQuery} $container
+   */
+  stopPageFlip($container){
+    const $svg = $container.find('svg');
+    $svg.on("mousedown", (event) => {
+      if ($(event.target).is('tspan')) {
+        event.stopPropagation();
+        $container.one("mouseup", (event) => event.stopPropagation());
+      }
+    });
+  }
+
+  /**
+   * @param {JQuery} $container
+   */
+  hideTextLayer($container) {
+    const $svg = $container.find('svg');
+    if ($container.hasClass("BRpageFlipping")) {
+      $svg.css('display', 'none');
+    }
+  }
+
+  /**
+   * @param {JQuery} $container
+   */
+  showTextLayer($container) {
+    const $svg = $container.find('svg');
+    if (!$container.hasClass("BRpageFlipping")) {
+      $svg.css('display', 'block');
+    }
+  }
+
+  /**
+   * @param {number} pageIndex
+   * @param {JQuery} $container
+   */
+  async createTextLayer(pageIndex, $container) {
+    const $svgLayers = $container.find('textSelctionSVG');
+    if (!$svgLayers.length) {
```

Comment:
Let's switch to a bail, so we can unindent the whole body.
```suggestion
    if ($svgLayers.length) return;
```

Reply:
done